openmdao/components/demux_comp.py

```diff
@@ -34,6 +35,9 @@ def __init__(self, **kwargs):
 
         self._no_check_partials = True
 
+        warn_deprecation("DemuxComp is being deprecated. This same functionality can be achieved "
+                         "directly in the connect/promotes indices arg using om.slicer.")
```

Comment:
I feel like this needs an example.  We are going to need a section in the docs that shows usage of Demux and what the replacement looks like with indices and the slicer.  I know that we will have to wait for that part until the docs are migrated back here.

Reply:
I'll put a hold on this PR until the doc is ready